generators/python/src/fern_python/generators/pydantic_model/type_declaration_handler/enum_generator.py

```diff
@@ -44,17 +45,24 @@ def generate(self) -> None:
                 should_export=True,
             )
         else:
-            enum_class = AST.ClassDeclaration(
-                name=self._get_class_name(),
-                extends=[
+            # If using python 3.11+, generate classes with StrEnum instead of (str, Enum)
+            if sys.version_info >= (3, 11):
```

Comment:
@jochs I think we'll want to discriminate based on this [config](https://github.com/fern-api/fern/blob/9274b21748ccb50183f654e12c86241382305ffb/generators/python/src/fern_python/generators/sdk/custom_config.py#L67)

Reply:
why? that already sets up the pyproject.toml -- does that not determine the env that the generator runs in? Happy to plumb it but it doesn't quite feel right to me since that config is a version string for a solver, not true version info

Reply:
ugh I see that's just for the generated project... will this even work if the generator is running via 3.9? I'm not sure how the reference imports work (if it's just straight up string concat or not) but StrEnum isn't in the stdlib until 3.11 so if it relies on the presence of the thing we're kinda SOL